Tasks/ANT/anttask.ts

```diff
@@ -208,7 +233,14 @@ async function doWork() {
                 var envName = "JAVA_HOME_" + jdkVersion.slice(2) + "_" + jdkArchitecture.toUpperCase();
                 specifiedJavaHome = tl.getVariable(envName);
                 if (!specifiedJavaHome) {
-                    throw new Error('Failed to find specified JDK version. Please make sure environment variable ' + envName + ' exists and is set to the location of a corresponding JDK.');
+                    if (isWindows) {
+                        // attempt to discover java home property from registry on Windows
+                        specifiedJavaHome = readJavaHomeFromRegistry(jdkVersion, jdkArchitecture);
+                    } 
```

Comment:
Would it be better to have a method findJavaHome() which includes the logic of checking variable and then check registry if windows? That method can go into a common module and be used by all the Java tasks.